src/hotspot/share/services/mallocHeader.hpp

```diff
@@ -115,15 +122,20 @@ class MallocHeader {
   void set_footer(uint16_t v)       { footer_address()[0] = v >> 8; footer_address()[1] = (uint8_t)v; }
 
  public:
-
+  MallocHeader(const MallocHeader&) = default;
   inline MallocHeader(size_t size, MEMFLAGS flags, uint32_t mst_marker);
 
   inline size_t   size()  const { return _size; }
   inline MEMFLAGS flags() const { return (MEMFLAGS)_flags; }
   inline uint32_t mst_marker() const { return _mst_marker; }
   bool get_stack(NativeCallStack& stack) const;
 
+  // Return the necessary data to record the block this belongs to as freed
+  FreePackage free_package() {
+    return FreePackage{this->size(), this->flags(), this->mst_marker()};
+  }
   inline void mark_block_as_dead();
+  inline void mark_block_as_alive();
```

Comment:
Proposal: name this "revive" and assert for dead markers. So far this is a one-trick pony used in realloc(), we don't really want a general way to mark dead headers as life.